app/src/androidTest/java/org/shadowice/flocke/andotp/ApplicationTest.java

```diff
@@ -168,22 +174,29 @@ public void testEntryURL() throws Exception {
         }
 
         Entry entry = new Entry("otpauth://totp/ACME%20Co:john.doe@email.com?secret=HXDMVJECJJWSRB3HWIZR4IFUGFTMXBOZ&issuer=ACME%20Co&ALGORITHM=SHA1&digits=6&period=30");
-        assertEquals("ACME Co - ACME Co:john.doe@email.com", entry.getLabel());
+        assertEquals("john.doe@email.com", entry.getLabel());
+
+        Entry entry2 = new Entry("otpauth://totp/ :john.doe@email.com?secret=HXDMVJECJJWSRB3HWIZR4IFUGFTMXBOZ&ALGORITHM=SHA1&digits=6&period=30");
+        assertEquals(":john.doe@email.com", entry2.getLabel());
+
+        Entry entry3 = new Entry("otpauth://totp/ :john.doe@email.com?secret=HXDMVJECJJWSRB3HWIZR4IFUGFTMXBOZ&issuer=%20&ALGORITHM=SHA1&digits=6&period=30");
+        assertEquals("john.doe@email.com", entry3.getLabel());
 
         assertEquals("HXDMVJECJJWSRB3HWIZR4IFUGFTMXBOZ", new String(new Base32().encode(entry.getSecret())));
 
 
         entry = new Entry("otpauth://totp/ACME%20Co:john.doe@email.com?secret=HXDMVJECJJWSRB3HWIZR4IFUGFTMXBOZ&issuer=ACME%20Co&ALGORITHM=SHA1&digits=6&period=30&tags=test1&tags=test2");
-        assertEquals("ACME Co - ACME Co:john.doe@email.com", entry.getLabel());
+        assertEquals("john.doe@email.com", entry.getLabel());
 
         assertEquals("HXDMVJECJJWSRB3HWIZR4IFUGFTMXBOZ", new String(new Base32().encode(entry.getSecret())));
         String[] tags = new String[]{"test1", "test2"};
         assertEquals(tags.length, entry.getTags().size());
         assertTrue(Arrays.equals(tags, entry.getTags().toArray(new String[entry.getTags().size()])));
     }
 
+    @Test
     public void testSettingsHelper() throws KeyStoreException, CertificateException, NoSuchAlgorithmException, IOException {
-        Context context = getContext();
+        Context context = InstrumentationRegistry.getTargetContext();
```

Comment:
This is deprecated, use `InstrumentationRegistry.getInstrumentation().getTargetContext();` from `androidx.test.platform.app.InstrumentationRegistry` (see comment above).